__init__.py

```diff
@@ -476,6 +482,41 @@ def on_pairing_end(self, error_dialog):
         self.state = SetupState.INACTIVE
 
     # Pairing GUI events
+    ### Language Selection & Welcome Screen
+    def handle_welcome_screen(self):
+        self.state = SetupState.WELCOME
+        self.handle_display_manager("Welcome")
+        sleep(0.3) # Wait for display to show the screen before speaking
+        self.speak_dialog("welcome_screen", wait=True)
+        sleep(0.3) # Wait for display a bit before showing the next screen
+        self.handle_language_menu()
+
+    def handle_language_menu(self):
+        self.state = SetupState.SELECTING_LANGUAGE
+        # Name: display name to display in UI
+        # Code: Used by ovos shell locale, get tts and stt engines
+        # System Code: Used by system as full code is required
+        # This cannot come from skill settings as it does not support this format needed
```

Comment:
this comment is not correct, skill settings can have arbitrary dictionaries? you can use that exact same data format under `self.settings["langs"]`